compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -572,6 +570,24 @@ private boolean isCompileTimeAnnotationValue(BLangExpression expr) {
                 return true;
             case TYPE_CONVERSION_EXPR:
                 return isCompileTimeAnnotationValue(((BLangTypeConversionExpr) expr).expr);
+            case STATEMENT_EXPRESSION:
+                BLangStatementExpression stmtExpr = (BLangStatementExpression) expr;
+                List<BLangStatement> stmts = ((BLangBlockStmt) stmtExpr.stmt).stmts;
+
+                if (!((BLangLocalVarRef) stmtExpr.expr).varSymbol.name.value.startsWith(DESUGARED_MAPPING_CONSTR_KEY)) {
```

Comment:
What does this check do? The `DESUGARED_MAPPING_CONSTR_KEY` constant doesn't seem to be set as a var name anywhere?

Reply:
This is the name we use when desugaring the record literal to a struct or map literal - https://github.com/ballerina-platform/ballerina-lang/blob/133d9010951cf306ffc708243a18bc3e9f111c22/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L6172